activerecord/lib/active_record/relation/batches.rb

```diff
@@ -122,24 +122,97 @@ def find_in_batches(begin_at: nil, end_at: nil, batch_size: 1000, start: nil)
         end
       end
 
+      in_batches(of: batch_size, begin_at: begin_at, end_at: end_at, load: true) do |batch|
+        yield batch.to_a
+      end
+    end
+
+    # Yields ActiveRecord::Relation objects to work with a batch of records.
+    #
+    #   Person.where("age > 21").in_batches do |relation|
+    #     sleep(50)
+    #     relation.update_all(cool: true)
+    #   end
+    #
+    # If you do not provide a block to #in_batches, it will return an Enumerator
+    # which yields ActiveRecord::Relation objects for chaining with other methods:
+    #
+    #   Person.in_batches.with_index do |relation, batch|
+    #     puts "Processing relation ##{batch}"
+    #     relation.each{|relation| relation.delete_all }
+    #   end
+    #
+    # ==== Options
+    # * <tt>:of</tt> - Specifies the size of the batch. Default to 1000.
+    # * <tt>:load</tt> - Specifies if the relation should be loaded. Default to false.
+    # * <tt>:begin_at</tt> - Specifies the primary key value to start from, inclusive of the value.
+    # * <tt>:end_at</tt> - Specifies the primary key value to end at, inclusive of the value.
+    #
+    # This is especially useful if you want to work with the
+    # ActiveRecord::Relation object instead of the individual records.
+    #
+    # This is especially useful if you want multiple workers dealing with
```

Comment:
`This is especially useful` is used twice in short order. Lets rephrase one of them.